src/watchmaker/workers/salt.py

```diff
@@ -262,10 +273,47 @@ def _build_salt_formula(self, extract_dir):
                 salt_content_filename
             ))
             self.retrieve_file(self.salt_content, salt_content_file)
+            temp_extract_dir = extract_dir
+            if self.salt_content_path:
+                temp_extract_dir = os.sep.join((
+                    self.working_dir,
+                    'salt-archive'
+                ))
             self.extract_contents(
                 filepath=salt_content_file,
-                to_directory=extract_dir
+                to_directory=temp_extract_dir
             )
+            if self.salt_content_path:
+                self.log.debug(
+                    'Using salt content path: %s',
+                    self.salt_content_path
+                )
+                salt_content_src = os.sep.join((
+                    temp_extract_dir, self.salt_content_path))
+                salt_files_dir = None
+                for (i, path) in enumerate(glob.glob(salt_content_src)):
+                    if os.path.isdir(path):
+                        if i > 0:
+                            msg = 'Found multiple paths matching' \
+                                  ' \'{0}\' in {1}'.format(
+                                      self.salt_content_path,
+                                      self.salt_content)
+                            self.log.critical(msg)
+                            raise WatchmakerException(msg)
+                        salt_files_dir = path
+                self.log.debug('Using extracted content from: %s',
+                               salt_files_dir)
+
+                for subdir in next(os.walk(salt_files_dir))[1]:
+                    if (
+                            not subdir.startswith('.') and
```

Comment:
this all seems indented a bit much...
```suggestion
                        not subdir.startswith('.') and
                        not os.path.exists(os.sep.join((extract_dir, subdir)))
```

Reply:
Yup, that indent needs to go.